packages/react-batch/README.md

```diff
@@ -1,3 +1,58 @@
 # react-batch
 
-Exports [boxart](github.com/boxart/boxart)'s Batch and BatchFactory. If using more of boxart use boxart-batch instead. Tools for building html games with React.
+Exports [boxart](https://github.com/boxart/boxart)'s Batch and BatchFactory. If using more of boxart use boxart-batch instead. Tools for building html games with React.
+
+## usage
+
+`<Batch>` and `<BatchFactory>` make optimization assumptions on passed item arrays so that element creation and re-renders are reduced. If the items passed compared with the last props are through a shallow compare unchanged the underlying react virtual element isn't recreated (saving some time, object creation and garbage collection) and the object won't be re-rendered (saving further time, object creation, and garbage collection).
+
+Their use is to replace mapping an array of data to React elements.
+
+```js
+{items.map(item => <ItemComponent item={item} />)}
+```
+
+### Batch
+
+Mapping an array is replaced with a Batch component, taking the properties `items`, `itemKey`, `children` and `component`.
+
+- `items` is simply the array whose rendering it is helping to optimize.
+- `itemKey` is a required handle to produce a key to represent a member of the items array.
+- `children` is our handler for creating components for members of the items array.
+- `component` specifies the tag wrapping the rendered items. It defaults to `'div'`.
+
+```js
+<Batch items={items} itemKey={item => item.key}>{(
+  item => <ItemComponent item={item} />
+)}</Batch>
+```
+
+`children` in most React uses are a set of deeper React elements. Instead Batch like react-intl uses the property to produce the representations of members of the items array. This handle is called for each member of items when Batch is first created and everytime a member changes. Batch does not rebuild the rendered elements if `children` changes, so its best to not pass a different function to the same batch.
```

Comment:
I stumbled on this... Maybe more like:

----

Unlike other react components which expect elements as children (contents inside the tag), `<Batch>` expects a function to be passed as `children` as in the example: `<Batch {...batch}>{ item => <ItemComponent {...item} /> }</Batch>`.  This function is called whenever Batch renders an item (when the item changes, or is first rendered). Batch **does not** rebuild the rendered elements if `children` changes, so its best to not pass a different function to the same batch.